### ISSUE TITLE
Support Tagging for IAM OIDC Providers

### ISSUE THREAD

**Issue by pshuman-heb** [hashicorp/terraform-provider-aws#18108]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

The `aws_iam_openid_connect_provider` resource doesn't currently support tags, while the underlying resource in AWS does. It seems like there's a field to tag them on [creation in the API](https://docs.aws.amazon.com/IAM/latest/APIReference/API_CreateOpenIDConnectProvider.html), and there's also a separate [API endpoint](https://docs.aws.amazon.com/IAM/latest/APIReference/API_TagOpenIDConnectProvider.html) that can add tags to them once created. Being able to tag these as part of our Terraform, instead of manually later, would be useful.

### New or Affected Resource(s)

* aws_iam_openid_connect_provider

### Potential Terraform Configuration

```hcl
resource "aws_iam_openid_connect_provider" "default" {
  url = "https://accounts.google.com"

  client_id_list = [
    "266362248691-342342xasdasdasda-apps.googleusercontent.com",
  ]

  thumbprint_list = []

  tags = {
    foo = true
    bar = "stuff"
  }
}
```

### References

* https://docs.aws.amazon.com/IAM/latest/APIReference/API_CreateOpenIDConnectProvider.html
* https://docs.aws.amazon.com/IAM/latest/APIReference/API_TagOpenIDConnectProvider.html

**Comment by pshuman-heb** [comment]
Potentially will be solved by #17964

**Comment by ewbankkit** [comment]
@pshuman-heb Thanks for raising this issue.
It has already been noticed in #17594. I'm going to close this one as a duplicate so that we can concentrate discussion in the linked issue.
Please add any additional comments there.
